### DISCUSSION TITLE
Keypad não funciona

### DISCUSSION THREAD

**Discussion by lpsouzalima** [uncodead/BrewUNO#79] [Q&A]
Olá.

Sou usuário do BrewUno (firmware_1.0.16) e atualmente utilizo o Wemos D1 Mini, com o controlador PCF8574P na mesma placa de interface criada por mim.

Tudo funciona perfeitamente, ja validei se as ligações estão corretas, e mesmo assim o teclado não responde.

Instalei um sketch que faz teste do teclado com o LCD e está funcionando a comunicação perfeitamente(esse sketch anexado e para um keypad 3x4 mas da para testar a comunicação do 1x4). Comparando com o código do brewuno vi que utiliza bilbiotecas diferentes para comunicação com o teclado e LCD.

Consegue me dar uma ajuda?

Obrigado!

[sketch_keypad&lcd.txt](https://github.com/uncodead/BrewUNO/files/6408177/sketch_keypad.lcd.txt)

**Comment by uncodead** [comment]
Tem que dar uma olhada na PCF e ver qual endereço dela

Em sex., 30 de abr. de 2021 às 18:35, lpsouzalima ***@***.***>
escreveu:

**Reply by lpsouzalima** [reply]
Executei I2C Scanner e encontrei dois endereços, um do display(funcionando) e do PCF(não funcionando).

I2C device found at address 0x20  -> PCF8574P
I2C device found at address 0x27  -> DISPLAY

Na configuração do brewuno vi que está disponível a alteração do endereço do PCF, mas a configuração esta apontando justamente para o do PCF e mesmo assim não funciona.

**Reply by lpsouzalima** [reply]
E ai amigo, consegue me ajudar?

**Reply by lbaraldi** [reply]
o PCF de endereço 0x20, é sua expansão correto? Você conectou a interrupção entre ele e o Wemos? Usou o resistor PullUP recomendado no esquema elétrico do PCF? 

O BrewUno usa interrupção para ler o o teclado. O Código é condicionado a uma interrupção do PCF para somente então "olhar" se alguma tecla foi pressionada.
Se não estiver devidamente implementada a ligação de interrupção na sua shield, o teclado não vai funcionar mesmo....

**Reply by lpsouzalima** [reply]
Exatamente, o endereço 0x20 e da minha extensão. A ligação entre o wemos e o pcf e direta através da saída gpio0 do wemos com a porta INT do PCF.
Não coloquei o pull up do wemos de início mas de qualquer forma a tensão saída gpio0 está em 3,3v, logo entendo que ela está UP e não precisaria do resistor.
No PCF sim tem os pullup nas portas sda e scl não tem nada na porta INT .
Após sua orientação sobre o pull up eu coloquei um na ligação entre o wemos e o PCF e TB não tive resultado.

Medi a tensão nos pinos do keypad e sempre que e acionado a tensão cai de 5v para 0v, indicando que o keypad está sendo acionado, e a interrupção que vc mencionou seria essa?
